src/Orleans.Core/Runtime/GrainInterfaceMap.cs

```diff
@@ -132,14 +132,6 @@ internal void AddMap(GrainInterfaceMap map)
                     directoriesIndex.Add(kvp.Key, kvp.Value);
                 }
             }
-
-            if (map.grainTypeTypeCodeMap is not null)
```

Comment:
I think we still need to add it somehow, no?

Reply:
Discussed offline: refactored code to not store `Type` and only uses built-in c# types. Now `GrainReference` does not create its own dictionary and only maps to use `GrainInterfaceMap`'s data